py/src/nupic/encoders/hexy.py

```diff
@@ -0,0 +1,235 @@
+# ------------------------------------------------------------------------------
+# Copyright (c) 2017 Norbu Tsering.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+# Source:  https://github.com/RedFT/Hexy
```

Comment:
we need to do a better job at separating 3rd party libs. 
- @dkeeney can you fetch this in cmake configuration phase and put it in some expected location? 
- otherwise, at lease place it in `/py/external/` 
- from its git, hexy needs to call a "pip install requirements.txt", @ctrl-z-9000-times seems we need that, unless we use only a limited subset that is covered here

Reply:
> from its git, hexy needs to call a "pip install requirements.txt", 

I don't think we need that stuff...  The hexy library contains a bunch of extraneous stuff for making & viewing hex maps, but the file I need only imports numpy.

> otherwise, at lease place it in /py/external/

+1

Reply:
> @dkeeney can you fetch this in cmake configuration phase and put it in some expected location?

If I remember correctly, you said that you had to make a few tweaks to get this to work.  So downloading is not a optioon.  So I will move it to Externals and setup a compile.

I should do that in a separate PR rather than try to mess with this one.

Reply:
Oh, this is the .py code.  I was thinking about the hash routine you were using.
I will put it into py/external/

Reply:
Ok, I moved the hexy library to `py/src/nupic/external/` which should be good enough for now.  It would be great if this was downloaded either from git or better yet pypi, but it's not critical.

Reply:
If pip has the package, then you could add that to bindings/py/packaging/requirements.txt

Reply:
bump, please prefer to use `pip` version, see https://github.com/htm-community/nupic.cpp/pull/353#discussion_r272038512